### ISSUE TITLE
Integration calls `async_track_state_change` instead of `async_track_state_change_event` which is deprecated

### ISSUE THREAD

**Issue by KrzysztofHajdamowicz** [soloam/ha-pid-controller#54]
> Detected that custom integration 'pid_controller' calls `async_track_state_change` instead of `async_track_state_change_event` which is deprecated and will be removed in Home Assistant 2025.5 at custom_components/pid_controller/sensor.py, line 831: async_track_state_change(self.hass, entity, sensor_state_listener), please create a bug report at https://github.com/soloam/ha-pid-controller/issues

**Comment by hauard** [comment]
@soloam Do you mind taking a look at this? Closing up on 2025.5 soon

**Comment by KrzysztofHajdamowicz** [comment]
@soloam ping

**Comment by soloam** [comment]
Personal life and work didn't allow me to dedicate the time to this! But I see that this integration, with the issues that are present, still has a lot of traction and people using it! I'll try to get time to solve this issue!

**Comment by egaldg** [comment]
yeah would be really great and appreciated ! i am new to using this nice integration and spending a lot of time on HA to implement this :-) so would be a bummer if it stopped working...

**Comment by soloam** [comment]
I need to undestand, it was launched on HACS a new version of a PID controller, probably someone with more time that me :)

https://github.com/bvweerd/simple_pid_controller

Does it still makes sence to upgrade this one? Or it's better to depracate this one ane route evry one to the new one?

**Comment by KrzysztofHajdamowicz** [comment]
It would be a pain for me to migrate to new integration as your work is a vital part of my setup.
Reconfiguring new one to accept new values range during the day and night would pose some troubles.

Now my devices have separate output ranges for day and night and automation changes required values.

**Comment by imLeeat** [comment]
same here, would be great if your PID keeps working with HA... dont want to migrate to a other integration.
It is really working well, so it would be awesome if you just update it a little bit to keep it working...
